Failure to build with GHC 8

```
Building cgen-0.0.5...
Preprocessing library cgen-0.0.5...
Preprocessing executable 'cgen' for cgen-0.0.5...
[1 of 8] Compiling DeriveMod        ( src/DeriveMod.hs, dist/build/cgen/cgen-tmp/DeriveMod.dyn_o )

src/DeriveMod.hs:59:14: error:
    • Ambiguous type variable ‘t0’ arising from a use of ‘concatMap’
      prevents the constraint ‘(Foldable t0)’ from being solved.
      Relevant bindings include
        fs :: t0 Name (bound at src/DeriveMod.hs:58:3)
      Probable fix: use a type annotation to specify what ‘t0’ should be.
      These potential instances exist:
        instance Foldable (Either a) -- Defined in ‘Data.Foldable’
        instance Foldable Maybe -- Defined in ‘Data.Foldable’
        instance Foldable ((,) a) -- Defined in ‘Data.Foldable’
        ...plus one other
        ...plus 23 instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the expression: concatMap crf fs
      In an equation for ‘exps’: exps = concatMap crf fs
      In the expression:
        do { fs <- dToFs d;
             let exps = concatMap crf fs;
             return $ exps }

src/DeriveMod.hs:64:11: error:
    • The constructor ‘DataD’ should have 6 arguments, but has been given 5
    • In the pattern: DataD _ _ _ cons _
      In the pattern: TyConI (DataD _ _ _ cons _)
      In a stmt of a 'do' block: TyConI (DataD _ _ _ cons _) <- reify d
note: keeping build directory ‘/tmp/nix-build-cgen-0.0.5.drv-2’
builder for ‘/nix/store/y3cw0p32c3iyxdb7h8c8l5ldyzxqsmhz-cgen-0.0.5.drv’ failed with exit code 1
```

--- a/src/DeriveMod.hs
+++ b/src/DeriveMod.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 module DeriveMod(deriveMod, deriveSMod, deriveMods, modify)
 where
 
@@ -61,7 +62,11 @@ mkModM crf d = do
 
 -- get the fields of a data type.
 dToFs d = do
-  TyConI (DataD _ _ _ cons _) <- reify d
+#if (__GLASGOW_HASKELL__ > 710)
+  TyConI (DataD _ _ _ _ cons _) <- reify d
+#else
+  TyConI (DataD   _ _ _ cons _) <- reify d
+#endif
   return $ concatMap getF cons
 
 deriveMods :: Name -> Q [Dec]

